packages/api/src/command/medical/patient/append-doc-query-progress.ts

```diff
@@ -48,52 +60,161 @@ export async function appendDocQueryProgress({
+    } else {
+      documentQueryProgress = setDocQueryProgress(
+        documentQueryProgress,
+        downloadProgress,
+        convertProgress,
+        convertibleDownloadErrors,
+        increaseCountConvertible
+      );
     }
 
+    documentQueryProgress.requestId = requestId;
+
     const updatedPatient = {
       ...existingPatient,
       data: {
         ...existingPatient.data,
         documentQueryProgress,
       },
     };
-    await PatientModel.update(updatedPatient, { where: patientFilter, transaction });
+
+    await PatientModel.update(updatedPatient, {
+      where: patientFilter,
+      transaction,
+    });
+
+    await processDocQueryProgressWebhook({
+      patient: existingPatient,
+      documentQueryProgress,
+      requestId,
+    });
+
     return updatedPatient;
   });
 }
+
+export const setDocQueryProgress = (
+  documentQueryProgress: DocumentQueryProgress,
+  downloadProgress?: Progress | undefined | null,
+  convertProgress?: Progress | undefined | null,
+  convertibleDownloadErrors?: number,
+  increaseCountConvertible?: number
+): DocumentQueryProgress => {
+  if (downloadProgress) {
+    documentQueryProgress.download = {
+      ...documentQueryProgress.download,
+      ...downloadProgress,
+    };
+  } else if (downloadProgress === null) {
+    documentQueryProgress.download = undefined;
+  }
+
+  if (convertProgress) {
+    documentQueryProgress.convert = {
+      ...documentQueryProgress.convert,
+      ...convertProgress,
+    };
+  } else if (convertProgress === null) {
+    documentQueryProgress.convert = undefined;
+  }
+
+  const convert = documentQueryProgress.convert;
+
+  if (convert && convertibleDownloadErrors != null && convertibleDownloadErrors > 0) {
+    convert.total = Math.max((convert.total ?? 0) - convertibleDownloadErrors, 0);
+    // since we updated the total above, we should update the status as well
+    convert.status = getStatusFromProgress(convert);
+  }
+
+  if (convert && increaseCountConvertible != null && increaseCountConvertible !== 0) {
+    convert.total = Math.max(0, (convert.total ?? 0) + increaseCountConvertible);
+
+    convert.status = getStatusFromProgress(convert);
+  }
+
+  return documentQueryProgress;
+};
+
+type RequiredProgress = Required<Progress>;
+
+function aggregateDocProgress(hieDocProgresses: DocumentQueryProgress[]): {
+  download?: RequiredProgress;
+  convert?: RequiredProgress;
+} {
+  const tallyResults = hieDocProgresses.reduce(
+    (acc: { download?: RequiredProgress; convert?: RequiredProgress }, progress) => {
+      const statuses: DocumentQueryStatus[] = [];
+
+      for (const type of progressTypes) {
+        const progressType = progress[type];
+
+        if (progressType) {
+          const currTotal = progressType.total ?? 0;
+          const currErrors = progressType.errors ?? 0;
+          const currSuccessful = progressType.successful ?? 0;
+          const accType = acc[type];
+          statuses.push(progressType.status);
```

Comment:
Yea this does work and i tested it via the unit tests. The idea is that for every status added we will aggregate again. Instead of doing this in the reducer though i can just create another function to aggregate teh status after the fact